server/sonar-process-monitor/src/main/java/org/sonar/application/AppStateListener.java

```diff
@@ -17,19 +17,18 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.server.platform.cluster;
+package org.sonar.application;
 
-import org.junit.Test;
-import org.sonar.test.TestUtils;
+import org.sonar.process.ProcessId;
 
-import static org.assertj.core.api.Assertions.assertThat;
-
-public class ClusterPropertiesTest {
-
-  @Test
-  public void test_Definitions() {
-    assertThat(ClusterProperties.definitions()).isNotEmpty();
-    assertThat(TestUtils.hasOnlyPrivateConstructors(ClusterProperties.class)).isTrue();
-  }
+@FunctionalInterface
+public interface AppStateListener {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Drop this interface in favor of "java.util.function.Consumer<ProcessId>". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1711)